### PR TITLE
Update Readme with release information

### PR DESCRIPTION
**Pull request by carlosalberto**

### CODE REVIEW

**Comment by jkwatson** [comment]
We should also update the Summary section, as it's a little out of date at this point.

**Comment by jkwatson** [comment]
@carlosalberto can you add the OTLP exporter? Then I think we can merge this.

**Comment by carlosalberto** [comment]
@jkwatson Done. You have something specifying in mind for the Summary section? Else we can merge this one now (and later update it in a follow-up PR).

File: README.md [hunk 1]
```diff
@@ -69,38 +69,34 @@ dependencies {
...
-- Additions (into interfaces for instance) will involve a best-effort attempt at
-  backwards compatibility (again, for implementations – callers of the public
-  APIs should not be negatively affected by these additions).
+## Releases
+
+OpenTelemetry Java is under active development. Releases aren't guaranteed
+to conform to a specific version of the specification. Future releases will
+not attempt to maintain backwards compatibility with previous releases.
+
+The latest version of the library was released on March 27th, 2020.
```
Comment (jkwatson) [inline README.md:78]:
```suggestion
The latest version of the libraries were released on March 27th, 2020.
```

**Comment by jkwatson** [comment]
> @jkwatson Done. You have something specifying in mind for the Summary section? Else we can merge this one now (and later update it in a follow-up PR).

The summary cleanup can happen in a follow-on PR, for sure.

**Pull request merged**